Dockerfile

```diff
@@ -6,70 +6,71 @@
 # Pull base image.
 FROM phusion/baseimage:0.9.22
 
-# Set correct environment variables.
+LABEL name="Check If You Can Get Legal Aid (cla_public)" \
+      maintainer="LAA Get Access <laa-get-access@digital.justice.gov.uk>" \
+      version="1.0"
+
 ENV HOME /root
-# Use baseimage-docker's init process.
-CMD ["/sbin/my_init"]
 
-# Dependencies
-RUN DEBIAN_FRONTEND='noninteractive' apt-get update && \
-  apt-get -y --force-yes install apt-utils python-pip \
-  python-dev build-essential git software-properties-common \
-  python-software-properties libpq-dev g++ make libpcre3 libpcre3-dev libffi-dev \
-  nodejs npm tzdata
+# Install python and build packages
+RUN curl -sL https://deb.nodesource.com/setup_8.x | bash - && \
+    apt-get -y --force-yes install \
+      apt-utils \
+      build-essential \
+      g++ \
+      git \
+      libffi-dev \
+      libpcre3 \
+      libpcre3-dev \
+      libpq-dev \
+      make \
+      nodejs \
+      python-dev \
+      python-pip \
+      python-software-properties \
+      software-properties-common \
+      tzdata
 
 # Set timezone
 RUN ln -fs /usr/share/zoneinfo/Europe/London /etc/localtime
 
-# Install Nginx.
-RUN DEBIAN_FRONTEND='noninteractive' add-apt-repository ppa:nginx/stable && apt-get update
-RUN DEBIAN_FRONTEND='noninteractive' apt-get -y --force-yes install nginx-full && \
-  chown -R www-data:www-data /var/lib/nginx
-
-RUN rm -f /etc/nginx/sites-enabled/default
-
-# Pip install Python packages
-RUN pip install -U setuptools pip wheel
-RUN pip install GitPython uwsgi
-
-RUN mkdir -p /var/log/wsgi && chown -R www-data:www-data /var/log/wsgi && chmod -R g+s /var/log/wsgi
-
-ADD ./docker/cla_public.ini /etc/wsgi/conf.d/cla_public.ini
-
-# install service files for runit
-ADD ./docker/uwsgi.service /etc/service/uwsgi/run
-
-# install service files for runit
-ADD ./docker/nginx.service /etc/service/nginx/run
+# Install nginx
+RUN add-apt-repository ppa:nginx/stable && \
+    apt-get update && \
+    apt-get -y --force-yes install nginx-full && \
+    chown -R www-data:www-data /var/lib/nginx && \
+    rm -f /etc/nginx/sites-enabled/default && \
+    mkdir -p /var/log/nginx/cla_public
+
+# Install uwsgi
+RUN pip install GitPython uwsgi && \
+    mkdir -p /var/log/wsgi && \
+    chown -R www-data:www-data /var/log/wsgi && \
+    chmod -R g+s /var/log/wsgi
+
+COPY ./docker/cla_public.ini /etc/wsgi/conf.d/cla_public.ini
+COPY ./docker/uwsgi.service /etc/service/uwsgi/run
+COPY ./docker/nginx.service /etc/service/nginx/run
+COPY ./docker/nginx.conf /etc/nginx/nginx.conf
 
-# Define mountable directories.
-#VOLUME ["/data", "/var/log/nginx", "/var/log/wsgi", "/var/log/cla_public"]
-RUN mkdir -p /var/log/nginx/cla_public
-
-# Expose ports.
-EXPOSE 80
-
-# APP_HOME
 ENV APP_HOME /home/app/flask
```

Comment:
Is this used anywhere?

Reply:
I am not sure. It sounds like it has a potential "dig through code" effect and as it does not add overhead to the Docker build I did not investigate. Could we defer it to a follow-up pull request?

Reply:
Sure